### PR TITLE
inequality and solveset modifications

### LINKED ISSUE #10214: solveset is not solving equations like sqrt(x) - const
There is an XFAIL test that indicates that solveset is doing something wrong: the following fails in master

```
solveset(x**(S(3)/2) + 4, x, S.Reals) == S.EmptySet
```

### PR DESCRIPTION
**Pull request by smichr**
closes #10214 
- [ ] make `solve_linear_inequality` private again (`_solve_inequality`)

### CODE REVIEW

File: sympy/solvers/inequalities.py [hunk 1]
```diff
@@ -395,109 +378,117 @@ def solve_univariate_inequality(expr, gen, relational=True):
...
-def _solve_inequality(ie, s):
-    """ A hacky replacement for solve, since the latter only works for
-        univariate inequalities. """
-    expr = ie.lhs - ie.rhs
-    try:
-        p = Poly(expr, s)
-        if p.degree() != 1:
-            raise NotImplementedError
-    except (PolynomialError, NotImplementedError):
+def solve_linear_inequality(ie, s):
```
Comment (aktech) [inline sympy/solvers/inequalities.py]: should this be added to the public API?

Comment (smichr) [inline sympy/solvers/inequalities.py]: It seems to have a well defined function that is different than solve_linear -- in the case of an inequality one can only do when the coefficient sign is known. Also, this function doesn't produce a set or relationalal involving oo like solve_univariate_inequality -- it just rearranges the inequality. I'm open to discussion, of course. Those were the thoughts that I had that motivated me making this public.

Comment (smichr) [inline sympy/solvers/inequalities.py]: On second thought it won't hurt to leave it private until it is really clear how the various functions for solving/handling inequalities should work.

Comment (aktech) [inline sympy/solvers/inequalities.py]: Sorry for the late reply, I almost forgot this.
I agree with the second thought, we can make it public, when we are sure of the stability of API.

**Comment by smichr** [comment]
stale

**Pull request closed without merging**